query_execution/PolicyEnforcerBase.hpp

```diff
@@ -163,6 +174,8 @@ class PolicyEnforcerBase {
   // Key = query ID, value = QueryManagerBase* for the key query.
   std::unordered_map<std::size_t, std::unique_ptr<QueryManagerBase>> admitted_queries_;
 
+  std::unordered_set<std::size_t> removed_query_ids_;
```

Comment:
Thanks @zuyu 
Apologies for being slow to review this PR. Can you please add the TODO about garbage collection on ``removed_query_ids_``, and I will merge this PR.